paddle/fluid/operators/activation_op.cc

```diff
@@ -604,21 +604,21 @@ class ActivationOpDoubleGrad : public framework::OperatorWithKernel {
 
   void InferShape(framework::InferShapeContext* ctx) const override {
     if (static_cast<int>(kDepValue) & static_cast<int>(kDepX)) {
-      if (ctx->HasOutput("DX")) {
+      if (HasOutputs("DX") && ctx->HasOutput("DX")) {
```

Comment:
Why do you add `HasOutputs("DX") `?

Reply:
See the ConvDoubleGradProtoMaker,  when one of DX's denpendences is non, dot not add 'DX' in the OpDesc. So, needs check here.